### ISSUE TITLE
 Publish the Dosbox-x on Flathub

### ISSUE THREAD

**Issue by LeandroStanger** [egrath/dosbox-x-flatpak#1]
https://github.com/flathub/flathub/wiki/App-Submission

**Comment by incognitte** [comment]
I would like to suggest that this repo can be merged into https://github.com/joncampbell123/dosbox-x as a auto builder for flatpak package. Once it is included in that repo, then it is easier to publish it on flathub as an officially supported package.

**Comment by egrath** [comment]
@joncampbell123 
Interested in merging the flatpak built into your official tree?

**Comment by joncampbell123** [comment]
@egrath Certainly. What changes are involved?

**Comment by egrath** [comment]
Essentially none, it's just a shell script which uses a json build description to fetch, build and package everything up. If everything is integrated smoothly it's also possible to automatically publish it over at flathub.org, the De-facto Appstore for Linux.

**Comment by joncampbell123** [comment]
That works for me. Will it be a patch or a pull request or...?

**Comment by egrath** [comment]
I think it would be the simplest solution to clone my repository and integrate the needed files into your source tree

**Comment by joncampbell123** [comment]
As someone who's not familiar with Flatpak, please indicate what files I should copy from your repo.

Also, when I do, I will need to rename build.sh to build-flatpak.sh to avoid conflicts with build.sh already in this repo.

**Comment by egrath** [comment]
The files you need are:
com.github.joncampbell123.DOSBox-X.appdata.xml: This is the Metadata file used by Flatpak/Flathub to build the Description for the App inside the store
com.github.joncampbell123.DOSBox-X.desktop: The XDG Desktop File used to run the application
com.github.joncampbell123.DOSBox-X.json: Describes how to fetch and build DOSBox-X
com.github.joncampbell123.DOSBox-X.png: Icon referenced in the XDG Desktop File
build.sh: To automate the build of the flatpaks

**Comment by incognitte** [comment]
I would like to suggest a structure like this to be added to dosbox-x repo for generating flatpak.

├── build-flatpak.sh <-- build.sh from this repo
├── build-scripts <-- Already exists in dosbox-x repo
│   └── flatpak <--- add a new folder specificly for flatpak generation. (the files are from this repo)
│       ├── com.github.joncampbell123.DOSBox-X.appdata.xml
│       ├── com.github.joncampbell123.DOSBox-X.desktop
│       ├── .....
│       ├── com.github.joncampbell123.DOSBox-X.json
│       └── com.github.joncampbell123.DOSBox-X.png
└── README.flatpak <-- a readme file on how to generate flatpak

what do you think?

**Comment by joncampbell123** [comment]
@incognitte Works for me. I'm only concerned whether the Flatpak files conflict with the existing source tree. If they don't, then I'll bring them in as-is. Can you provide a pull request laying out the files exactly as they should?

**Comment by incognitte** [comment]
Cool! Give me a while and I will work on the Pull Request with the appropriate structure. Some of the files in this repo need to be updated if we want to have it structured as mentioned [above](https://github.com/egrath/dosbox-x-flatpak/issues/1#issuecomment-618495331).

**Comment by incognitte** [comment]
@joncampbell123 I was not able to use git submodule to add @egrath flatpak repo to your dosbox-x repo. (I am not that good with git as you can imagine). So I forked dosbox-x repo and manually added the changes. Here it is: https://github.com/incognitte/dosbox-x please check it out and if you know a git trick to do submodules properly please implement them correctly, otherwise I can do Push Requ

**Comment by incognitte** [comment]
@joncampbell123 sorry for my delay.

I am having a little hard time and the reason is that I am not familiar enough with "git" and its commands. Let me try to describe what I am trying to achieve and maybe somebody who is a git master can help out.

Following the Flathub [App Submission](https://github.com/flathub/flathub/wiki/App-Submission#how-to-submit-an-app) these steps need to be taken:

1. Fork the [Flathub repository on GitHub](https://github.com/flathub/flathub)
2. Clone the fork:
    `git clone --branch=new-pr git@github.com:your_github_username/flathub.git`
3. Create a new branch with your app’s name:
    `git checkout -b your_app_name`
4. Add your app’s manifest to the branch, commit it and push the commit
5. Open a pull request against  the `new-pr` branch on Github
6. Your pull request will then be reviewed by the Flathub admins 

Once this process is done a new repo will be added to [flathub's GitHub list](https://github.com/flathub) with the same name as the app. In our case it will be probably called something like `com.github.joncampbell123.DOSBox-X`. As an example, see [LibreOffice repo](https://github.com/flathub/org.libreoffice.LibreOffice).

However to make sure that this newly generated repo will be kept up to date by the same mentainers of DOSBox-X, **ideally** it needs to be also part of [DOSBox-X Github repo](https://github.com/joncampbell123/dosbox-x) under `build-scripts/flatpak` subfolder. Once the DOSBox-X repo is updated with the latest push, the flathub repo also gets updated (specially useful for keeping release date and version updated).

And this is where I am struggling. I don't know how to generate such a structure with git. 

@LeandroStanger or @egrath may be able to help?
